### PR TITLE
WIP: Add hasProperty() assertion

### LINKED ISSUE #55: Add hasProperty() assertion
This assertion should work roughly similar to `hasAttribute()` and can then be used to check for properties that don't have dedicated assertions (like `hasValue()`).

### PR DESCRIPTION
**Pull request by seansellek**
This takes the first steps towards addressing #38 by adding a base `.hasProperty()` assertion as mentioned by @Turbo87.

The assertion supports providing a custom message to facilitate building shortcuts such as `.isChecked()`.

Resolves #55

### CODE REVIEW

**Comment by seansellek** [comment]
There seems to be a permissions issue surrounding Chrome in CI 😥

`You need to make sure that  /opt/google/chrome/chrome-sandbox is owned by root and has mode 4755`

**Comment by Turbo87** [comment]
@seansellek I'm currently on vacation, I'll have a look once I'm back (7 days)

**Comment by Turbo87** [comment]
@seansellek 👋  we've just merged https://github.com/simplabs/qunit-dom/pull/48, which should fix the Chrome issue. Can you rebase this PR?

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@
 npm-debug.log*
 yarn-error.log
 testem.log
+.vscode/
```
Comment (Turbo87) [inline .gitignore]: This stuff should be move to your global `.gitignore` file (see https://help.github.com/articles/ignoring-files/)

Only project specific files should be added to this file, but no individual editor config files etc.

**Comment by seansellek** [comment]
@Turbo87 sorry it took a bit for me to get back to this. Rebased, removed the .gitignore commit, and fixed a bug in the tests, getting coverage up to 100%.

File: lib/assertions.js [hunk 2]
```diff
@@ -72,6 +73,23 @@ export default class DOMAssertions {
     notFocused.call(this, message);
   }
 
+  /**
+   * Assert that the [HTMLElement][] or an [HTMLElement][] matching the
+   * `selector` contains the provided `property` and optionally checks
+   * if the property `value` matches the provided `value`.
+   *
+   * @param {string} property
+   * @param {object?} options
```
Comment (Turbo87) [inline lib/assertions.js]: I'm wondering if we should mimic the `hasAttribute` API instead which accepts `name, value?, message?` 🤔

Comment (scalvert) [reply lib/assertions.js]: +1 to this. There's a lot to be said for the API being consistent for like methods/functions.

File: API.md [hunk 3]
```diff
@@ -495,3 +495,18 @@ Assert that the `value` property of an [HTMLInputElement][] is empty.
...
+
+**Parameters**
+
+-   `property` **[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)**
+-   `options` **[object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object)?**
+
+**Examples**
+
+```javascript
+assert.dom('input#agreedToTOS').hasProperty('checked', { value: true });
```
Comment (Turbo87) [inline API.md]: as mentioned before, I think this should mimic the API that we have for `hasAttribute()`

**Comment by scalvert** [comment]
Thanks for pinging on this, @Turbo87. I've been pushing on it over the last little while, but haven't had as much time to get it through.

I've aligned it more to the `hasAttribute` signature, and am just working through the tests. I'll try to get this finished up over the next little while.

**Comment by Turbo87** [comment]
sweet, thanks for working on it :)

**Comment by Turbo87** [comment]
closing due to inactivity

**Pull request closed without merging**